src/Plugin/resource/Field/ResourceFieldCollection.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Plugin\resource\Field\ResourceFieldCollection.
+ */
+
+namespace Drupal\restful\Plugin\resource\Field;
+
+class ResourceFieldCollection implements ResourceFieldCollectionInterface {
```

Comment:
`ResourceFieldCollection` can be seen as a glorified array of `ResourceFieldInterface` objects. But with the ability to add methods to the collection.

It can hold _resource fields_ of any kind (even mixing types in a collection). A _resource field_ is every one of the entries of the mapping array in 7.x-1.x + the public name.